tools/cli/helpers/checkEnvironment.js

```diff
@@ -108,7 +108,8 @@ export async function getPhpVersion() {
 		}
 		return;
 	}
-	return res.stdout.toString().trim();
+	// Strip any `-1` or the like, as various sources (e.g. sury) indicate patched versions in this manner which confuses `semver.satisfies`.
+	return res.stdout.toString().trim().replace( /-\d.*/, '' );
```

Comment:
I'd lean toward something more general:
```suggestion
	return res.stdout.toString().trim().replace( /-.*/, '' );
```

For example, I looked through some logs and found that `8.1.22-he.0` is a valid version, as is `8.3.0-dev`.

Reply:
`8.3.0-dev` is a prerelease version, I was unsure about whether we should mess with that.

I don't know what `8.1.22-he.0` might be for.

Reply:
If the ideal is 8.2, but we allow 8.3, we should also allow 8.3-dev in my opinion.

As for `8.1.22-he.0` (and other ones with the `he.0` suffix), I'm not sure what it's from, but it was common in some logs I was perusing. It's a valid version, though, the same as any other dash suffix, so if we're going to allow some dash suffixes, why not all?

Reply:
Yeah, I think you're right.